.github/workflows/met-api-cd.yml

```diff
@@ -0,0 +1,68 @@
+name: MET API DEV CD
+
+
+on:
+  # Disable triggers for now.
+  # push:
+  #   branches:
+  #     - main
+  #   paths:
+  #     - "met-api/**"
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: "Environment (dev/test/prod)"
+        required: true
+        default: "dev"
+
+defaults:
+  run:
+    shell: bash
+    working-directory: ./met-api
+
+env:
+  APP_NAME: "met-api"
+  TAG_NAME: "dev"
+
+jobs:
+  forms-flow-bpm-cd-by-push:
+    runs-on: ubuntu-20.04
+
+    if: github.event_name == 'push' && github.repository == 'bcgov/met-public'
+    environment:
+      name: "dev"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login Openshift
+        shell: bash
+        run: |
+          oc login --server=${{secrets.OPENSHIFT4_LOGIN_REGISTRY}} --token=${{secrets.OPENSHIFT4_SA_TOKEN}}
+
+
+      - name: Tools project
+        shell: bash
+        run: |
+          oc project ${{ secrets.OPENSHIFT4_REPOSITORY }}-tools
+
+      # - name: Build from dev branch
+      #   shell: bash
+      #   run: |
+      #     oc patch bc/${{ env.APP_NAME }} -p '{"spec":{"source":{"git":{"ref":"dev"}}}}'
+
+      - name: Start Build Openshift
+        shell: bash
+        run: |
+          oc start-build ${{ env.APP_NAME }} --wait
+
+
+      - name: Tag+Deploy for DEV
+        shell: bash
+        run: |
+          oc tag ${{ env.APP_NAME }}:latest ${{ env.APP_NAME }}:${{ env.TAG_NAME }}
+
```

Comment:
for TEST or deploy workflows , we dont need to build the image
so we need to have a 
CD-push which can run on code merge
CD which can deploy things(just tag the images)

Reply:
I agree for now this will work (when we have a develop branch we will need to build the image based on the 'main' branch before deploying).
This is how it looks now:

- CD-push for each project but not in use yet (the trigger is currently disabled, openshift will build the images on push) when in use this would build and deploy to DEV

- Deploy Action that deploys the entire app (all images get tagged with either test or prod) this can be used to promote dev images into test/prod